### PR TITLE
Update ESR link on /welcome/20/ page to link to /firefox/all/desktop-esr/ (Fixes #15501)

### LINKED ISSUE #15501: /firefox/welcome/20/ page should link ESR users to /firefox/all/desktop-esr/
## Description

The https://www.mozilla.org/en-US/firefox/welcome/20/ page has some special code that detects Firefox ESR, and displays a different call to action for those users that points to `/firefox/enterprise/#download`. 

The issue here is that ESR users on unsupported operating systems, such as Windows 7/8/8.1, cannot install ESR 128. We should instead link ESR users to https://www.mozilla.org/firefox/all/desktop-esr/, where they have the ability to select ESR 115 as well as 128.

### PR DESCRIPTION
**Pull request by alexgibson**
## One-line summary

The folks working on the root-ca-succession project have agreed that this feels like the more appropriate link for ESR users.

## Issue / Bugzilla link

#15501

## Testing

- http://localhost:8000/en-US/firefox/welcome/19/
- http://localhost:8000/en-US/firefox/welcome/20/
- http://localhost:8000/en-US/firefox/welcome/21/
- http://localhost:8000/en-US/firefox/welcome/22/
- http://localhost:8000/en-US/firefox/welcome/23/
- http://localhost:8000/en-US/firefox/welcome/24/

### CODE REVIEW

File: bedrock/firefox/templates/firefox/welcome/page20.html [hunk 1]
```diff
@@ -42,7 +42,11 @@ <h1 class="c-main-title">{{ ftl('welcome-page20-21-your-firefox') }}</h1>
 
     {{ download_firefox_thanks(alt_copy=ftl('welcome-page20-21-update-now-cta'), button_class='mzp-t-xl', dom_id='update-firefox') }}
 
-    <p class="c-main-cta" id="update-firefox-esr"><a href="{{ url('firefox.enterprise.index') + '#download' }}" class="mzp-c-button mzp-t-product mzp-t-xl" data-cta-text="Update now">{{ ftl('welcome-page20-21-update-now-cta') }}</a></p>
+    <p class="c-main-cta" id="update-firefox-esr">
+      <a href="{{ url('firefox.all.platforms', product_slug='desktop-esr') }}" class="mzp-c-button mzp-t-product mzp-t-xl" data-cta-text="Update now">
```
[the remark this answers is not in the document]
Comment (alexgibson) [reply bedrock/firefox/templates/firefox/welcome/page20.html:46]: Ah I was not aware of this - thanks!

[the remark this answers is not in the document]
Comment (janbrasna) [reply bedrock/firefox/templates/firefox/welcome/page20.html:46]: TBH I'm trying to infer whether page20 is the only one targeted at ESR115-compatible OSes (as they're served based on some remote-settings distribution or version rules), but doesn't fee like it from the brief for all waves of rootocalypse moments:

1. October 15 -- Message Firefox desktop versions <= 101.
2. December 10 --  Message Firefox desktop and Android versions <= 127.
3. January 9 --  Message Firefox desktop versions <= 101 again.
4. (January 14 -- A scheduled server change will prevent us from further messaging users on Firefox desktop versions <= 101.)
5. February 5 --  Message Firefox desktop and Android versions 102-127 again.
6. March 11 --  Message Firefox desktop and Android versions 102-127 one last time.
7. March 14 -- The old root certificate expires! 🪦

[the remark this answers is not in the document]
Comment (alexgibson) [reply bedrock/firefox/templates/firefox/welcome/page20.html:46]: Irrespective I think this makes the most sense for ESR users as they get the choice of ESR 128 or ESR 115 either way.

[the remark this answers is not in the document]
Comment (alexgibson) [reply bedrock/firefox/templates/firefox/welcome/page20.html:46]: The issue only effects users who are already using Firefox ESR on outdated operating systems.

Will update welcome-19 link just to be safe as well 👍

**Review by stephaniehobson — APPROVED** [review]
Oh yes, much better. My friend got welcome19 on her ancient MacBook Air and I had to do a support call with her 🙃 She would have still been confused but it would have saved me time. (Turns out she can't run 115 either, but that's pretty clear from the message on /all)

R+

**Pull request merged**